ports/esp32s2/pcnt_handler.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 microDev
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "pcnt_handler.h"
+
+#include "py/runtime.h"
+#include "supervisor/shared/translate.h"
+
+#define PCNT_UNIT_ACTIVE    1
+#define PCNT_UNIT_INACTIVE  0
+
+static uint8_t pcnt_state[4];
+
+void pcnt_handler_init(pcnt_config_t* pcnt_config) {
+    // Look for available pcnt unit
+    for (uint8_t i = 0; i<=3; i++) {
+        if (pcnt_state[i] == PCNT_UNIT_INACTIVE) {
+            pcnt_config->unit = (pcnt_unit_t)i;
+            pcnt_state[i] = PCNT_UNIT_ACTIVE;
+            break;
+        } else if (i == 3) {
+            mp_raise_RuntimeError(translate("No PCNT unit free"));
```

Comment:
I would change this to "All PCNT units in use" to emphasize they have been used up, as opposed to the device not having any to start with.